packages/babel-helpers/src/helpers.js

```diff
@@ -413,6 +416,29 @@ helpers.interopRequireWildcard = template(`
   })
 `);
 
+helpers.specRequireInterop = template(`
+  (function (obj) {
+    if (obj && obj.__esModule) {
+      return obj;
+    } else {
+      var newObj = Object.create(null, {
+          default: {
+            value: obj,
+            writable: true,
+            enumerable: true
+          },
+          __esModule: {
```

Comment:
yeah, that's true. it was just a thought - it can probably be inferred with `Object.isFrozen` or checking the descriptor.